common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/RelatedWorks.scala

```diff
@@ -42,12 +50,110 @@ object RelatedWorks {
       precededBy = default,
       succeededBy = default
     )
-}
 
-case class RelatedWork(
-  work: IdentifiedWork,
-  relatedWorks: RelatedWorks
-)
+  type TokenizedPath = List[PathToken]
+  type PathToken = List[PathTokenPart]
+  type PathTokenPart = Either[Int, String]
+
+  /** Creates RelatedWorks from unnested and unsorted data.
+    *
+    * @param path Children of the work
+    * @param children Direct descendents of the work
+    * @param siblings Siblings of the work
+    * @param ancestors Ancestors of the work
+    */
+  def apply(path: String,
+            children: List[IdentifiedWork],
+            siblings: List[IdentifiedWork],
+            ancestors: List[IdentifiedWork]): RelatedWorks = {
+    val mainPath = tokenizePath(path)
+    val (precededBy, succeededBy) = siblings.sortBy(tokenizePath).partition {
+      work =>
+        tokenizePath(work) match {
+          case None => true
+          case Some(workPath) =>
+            tokenizedPathOrdering.compare(mainPath, workPath) >= 0
+        }
+    }
+    val parent = ancestors.sortBy(tokenizePath) match {
+      case head :: tail =>
+        Some(
+          tail.foldLeft(RelatedWork(head, RelatedWorks(partOf = Some(Nil)))) {
+            case (relatedWork, work) =>
+              RelatedWork(work, RelatedWorks.partOf(relatedWork))
```

Comment:
Would this create a `parent` that doesn't have it's `siblings` nor `children`?

Reply:
This is what has been requested from the API: `partOf` is nested but nothing else.

Reply:
Yep that's right.  This is what has been requested from the API: `partOf` is nested but nothing else.

Reply:
Ah, perfect, so you can traverse up, but then not down another path etc... thanks for clarifying.